### PR TITLE
Begin transition to WASM

### PR DESCRIPTION
**Pull request by ghost**
- [x] Remove yarn
- [x] Remove protodefc-prebuild
- [x] Add scripts for building native binaries
- [x] Compile and load
- [x] Add tests
- [x] Have https://github.com/ProtoDef-io/node-protodefc/issues/5#issuecomment-393732168 be completed first (i.e.: them both merged)

To build, npm run build, this will require rust nightly, wasm-bindgen and the wasm unknown target to be installed in your Rust toolchain. See https://github.com/rustwasm/wasm-bindgen#basic-usage for more information.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,2 +1,73 @@
-node_modules
-package-lock.json
\ No newline at end of file
+### Node ###
+# Logs
+logs
+*.log
+npm-debug.log*
+yarn-debug.log*
```
Comment (rom1504) [inline .gitignore]: Why do you add so much stuff we don't have in the repo ?
IMHO we should only have in gitignore things that are generated by us and we don't want to compile.
You can use the global git ignore file for other stuff

**Review by rom1504 — DISMISSED** [review]

File: .gitignore (hunk 1 above)
Comment (ghost) [reply .gitignore]: I think it's better to stay safe. If people contribute, we want them to know what we don't want in the repository.

**Comment by ghost** [comment]
Another thing that needs to be addressed in this PR is optimizations. Do we want to optimize for speed or size? I'm thinking size, as for those that want to run it in the browser, size is important. However, they should really prebuild their code with the CLI instead of requiring this module.

**Comment by ghost** [comment]
As well, error handling needs to be addressed. Currently, everything will just crash if there is any error anywhere. We need to fix this up.

**Comment by ghost** [comment]
Last item that is needed before this is merged is to have my pull requests at ProtoDefc merged

**Comment by ghost** [comment]
This now just needs to be locked to latest HEAD on ProtoDefc, and and then this can be merged. As well, we can then do a release of node-protodefc as 0.1.0 or something.

**Pull request merged**